packages/opencensus-core/src/stats/view.ts

```diff
@@ -181,15 +196,57 @@ export class BaseView implements View {
    * @param bucketBoundaries a list with the bucket boundaries
    */
   private createBuckets(bucketBoundaries: number[]): Bucket[] {
-    return bucketBoundaries.map((boundary, boundaryIndex) => {
-      return {
-        count: 0,
-        lowBoundary: boundaryIndex ? boundary : -Infinity,
-        highBoundary: (boundaryIndex === bucketBoundaries.length - 1) ?
-            Infinity :
-            bucketBoundaries[boundaryIndex + 1]
-      };
-    });
+    let negative = 0;
+    const result = bucketBoundaries.reduce((accumulator, boundary, index) => {
+      if (boundary >= 0) {
+        const nextBoundary = bucketBoundaries[index + 1];
+        this.validateBoundary(boundary, nextBoundary);
+        const len = bucketBoundaries.length - negative;
+        const position = index - negative;
+        const bucket = this.createBucket(boundary, nextBoundary, position, len);
+        accumulator.push(bucket);
+      } else {
+        negative++;
+      }
+      return accumulator;
+    }, []);
+    if (negative) {
+      this.logger.warn(`Dropping ${
+          negative} negative bucket boundaries, the values must be strictly > 0.`);
```

Comment:
ditto

Reply:
> ditto

This is what implemented in java [BucketBoundaries.java line 84](https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/stats/BucketBoundaries.java#L84)

Reply:
Ignore this comment - just noticed "${negative}" is already in the log.